clang/lib/Driver/Types.cpp

```diff
@@ -242,7 +242,7 @@ bool types::isCXX(ID Id) {
   case TY_CXXHUHeader:
   case TY_PP_CXXHeaderUnit:
   case TY_ObjCXXHeader: case TY_PP_ObjCXXHeader:
-  case TY_CXXModule: case TY_PP_CXXModule:
+  case TY_CXXModule: case TY_PP_CXXModule: case TY_ModuleFile:
```

Comment:
If this is needed, it should be in a separate patch. Let's do one thing in one PR.

Reply:
This is required to get rid of the `Wunused-argument: /std:c++20` warning.

I realised that the input file was `TY_ModuleFile` which led the Clang-CL driver to interpret it as a C file, and not a CXX file, and then eventually emitting this error.

Reply:
Fine enough. Please land this as a separate PR.

Reply:
Commits have been moved to #99300.